packages/core/src/config/parse.ts

```diff
@@ -1262,33 +1303,82 @@ export const parseStructConstructorArg = (
     )
   }
 
+  // We log some warnings if the user includes arrays in their struct constructor args since the Solidity compiler will ignore them
+  const arrayMembers = paramType.components
+    .map((el) => {
+      const fullParamType = {
+        paramType: ethers.utils.ParamType.from(el.type),
+        name: el.name,
+      }
+      return fullParamType
+    })
+    .filter((el) => el.paramType.baseType === 'array')
+
+  if (arrayMembers.length > 0) {
+    logValidationError(
+      'warning',
+      `Detected array(s) in struct constructor argument. Struct members that are arrays will be ignored by the Solidity compiler and cannot be set in the constructor.`,
+      [
+        'The following struct members will be ignored: ',
+        ...arrayMembers.map((el) => `${el.name}: ${el.paramType.type}`),
+      ],
+      cre.silent,
+      cre.stream
+    )
+  }
```

Comment:
I actually don't think this is necessary. The Solidity compiler doesn't ignore these values. I ran into this same issue a while ago, and it turns out that you just need to explicitly define a getter function for them in Solidity. See [here](https://github.com/chugsplash/chugsplash/blob/develop/packages/contracts/contracts/ChugSplashManager.sol#L740-L754) for an explanation (esp the linked github issue)

Reply:
Wow, that's pretty wonky

Reply:
Lol yep, welcome to Solidity